osquery/filesystem/filesystem.cpp

```diff
@@ -363,7 +408,7 @@ Status resolveFilePattern(std::vector<std::string> components,
 
 Status resolveFilePattern(const boost::filesystem::path& fs_path,
                           std::vector<std::string>& results) {
-  return resolveFilePattern(split(fs_path.string(), "/"), results, 0, 0);
+  return resolveFilePattern(split(fs_path.string(), "/"), results);
```

Comment:
The whole thing looks complicated but solid. Some of the wildcard checking/tokenizing/path Frankensteining might be clearer if broken out to intermediate methods. 

One note here, looks like you can take a relative path, then on 322, does that join clobber a relative into an absolute?